usaon_benefit_tool/models/tables.py

```diff
@@ -313,6 +313,58 @@ class AssessmentNode(BaseModel):
         back_populates="source_assessment_node",
     )
 
+    # HACK: make it look like node type field is local to this table to enable
+    # polymorphism.
+    # OPTIMIZE: This is going to cost an extra subquery as far as I can tell, could be
+    # expensive.
+    # TODO: Open an issue or something on sqlalchemy repo to understand better way to do
+    # this.
+    # TODO: If we can ever remove this hack, move __mapper_args__ back to top.
+    # See: https://stackoverflow.com/a/54934609
```

Comment:
This is the ugly part; SQLAlchemy doesn't like to do polymorphism on foreign keys. Implications documented in comments.